Docker build failing due to npm permissions

When running docker-compose build, I get an error message of:

=> [internal] load build definition from Dockerfile.oldev                                                         0.0s
 => => transferring dockerfile: 38B                                                                                0.0s
 => [internal] load .dockerignore                                                                                  0.0s
 => => transferring context: 34B                                                                                   0.0s
 => [internal] load metadata for docker.io/openlibrary/olbase:latest                                               0.9s
 => [1/6] FROM docker.io/openlibrary/olbase:latest@sha256:4d18fdc8a1efe96b565c5e99a096e1d69e132dbfd19bcb93bf34beb  0.0s
 => [internal] load build context                                                                                  0.0s
 => => transferring context: 150B                                                                                  0.0s
 => CACHED [2/6] WORKDIR /openlibrary                                                                              0.0s
 => CACHED [3/6] COPY requirements*.txt ./                                                                         0.0s
 => CACHED [4/6] RUN python3.9 -m pip install -r requirements_test.txt                                             0.0s
 => CACHED [5/6] COPY package*.json ./                                                                             0.0s
 => ERROR [6/6] RUN npm install                                                                                    6.2s
------
 > [6/6] RUN npm install:
#9 6.153 npm notice
#9 6.153 npm notice New minor version of npm available! 8.3.1 -> 8.5.2
#9 6.153 npm notice Changelog: <https://github.com/npm/cli/releases/tag/v8.5.2>
#9 6.153 npm notice Run `npm install -g npm@8.5.2` to update!
#9 6.153 npm notice
#9 6.154 npm ERR! code EACCES
#9 6.154 npm ERR! syscall open
#9 6.154 npm ERR! path /openlibrary/package-lock.json
#9 6.155 npm ERR! errno -13
#9 6.156 npm ERR! Error: EACCES: permission denied, open '/openlibrary/package-lock.json'
#9 6.156 npm ERR!  [Error: EACCES: permission denied, open '/openlibrary/package-lock.json'] {
#9 6.156 npm ERR!   errno: -13,
#9 6.157 npm ERR!   code: 'EACCES',
#9 6.157 npm ERR!   syscall: 'open',
#9 6.157 npm ERR!   path: '/openlibrary/package-lock.json'
#9 6.157 npm ERR! }
#9 6.157 npm ERR!
#9 6.157 npm ERR! The operation was rejected by your operating system.
#9 6.157 npm ERR! It is likely you do not have the permissions to access this file as the current user
#9 6.157 npm ERR!
#9 6.158 npm ERR! If you believe this might be a permissions issue, please double-check the
#9 6.158 npm ERR! permissions of the file and its containing directories, or try running
#9 6.158 npm ERR! the command again as root/Administrator.
#9 6.159
#9 6.159 npm ERR! A complete log of this run can be found in:
#9 6.159 npm ERR!     /home/openlibrary/.npm/_logs/2022-03-03T00_05_16_169Z-debug-0.log
------
failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/sh -c npm install]: exit code: 243

I'm guessing this is because I don't have access to the package-lock.json file, but how would I fix this? sudo and chmod wouldn't work for me bcz I'm running on windows and running it as admin did not work either.

--- a/docker/Dockerfile.olbase
+++ b/docker/Dockerfile.olbase
@@ -89,10 +89,10 @@ RUN mkdir -p /var/log/openlibrary /var/lib/openlibrary && chown openlibrary:open
 WORKDIR /openlibrary
 
 USER openlibrary
-COPY requirements*.txt ./
+COPY --chown=openlibrary:openlibrary requirements*.txt ./
 RUN python3.9 -m pip install --default-timeout=100 -r requirements.txt
 
-COPY package*.json ./
+COPY --chown=openlibrary:openlibrary package*.json ./
 RUN npm ci
 
 COPY --chown=openlibrary:openlibrary . /openlibrary

--- a/docker/Dockerfile.oldev
+++ b/docker/Dockerfile.oldev
@@ -1,8 +1,8 @@
 FROM openlibrary/olbase:latest
 WORKDIR /openlibrary
 
-COPY requirements*.txt ./
+COPY --chown=openlibrary:openlibrary requirements*.txt ./
 RUN python3.9 -m pip install -r requirements_test.txt
 
-COPY package*.json ./
+COPY --chown=openlibrary:openlibrary package*.json ./
 RUN npm install

